### PR TITLE
contract metadata v5

### PR DESCRIPTION
**Pull request by kziemianek**
Constructor and message metadata is going to contain new attribute named `default`. The main reason behind adding it is to allow UIs (ex. `contracts-ui`) determine which one constructor/message should be selected as default in lists.

This is follow-up pull request to https://github.com/paritytech/ink/issues/1703

### CODE REVIEW

**Comment by kziemianek** [comment]
I'm not sure if this should go to v4 or v5, finally I've decided to create v5.

File: packages/types/src/interfaces/contractsAbi/definitions.ts [hunk 1]
```diff
@@ -170,6 +184,16 @@ const spec = {
     returnType: 'Option<ContractTypeSpec>',
     docs: 'Vec<Text>'
   },
+  ContractMessageSpecV3: {
+    label: 'Text',
+    selector: 'ContractSelector',
+    mutates: 'bool',
+    payable: 'bool',
+    default: 'bool',
```
Comment (jacogr) [inline packages/types/src/interfaces/contractsAbi/definitions.ts:192]: This is not correct - we cannot introduce fields retrospectively to existing versions. In v3 “default” was not available, it is o lay there from v5+

Comment (kziemianek) [reply packages/types/src/interfaces/contractsAbi/definitions.ts:192]: Ok, thank you. I wasn't certain about versioning here.
Would something like this be ok ?
```
  ContractMessageSpecV3: 'ContractMessageSpecV2',
  ContractMessageSpecV4: 'ContractMessageSpecV2',
  ContractMessageSpecV5: {
    label: 'Text',
    selector: 'ContractSelector',
    mutates: 'bool',
    payable: 'bool',
    default: 'bool',
    args: 'Vec<ContractMessageParamSpecV2>',
    returnType: 'Option<ContractTypeSpec>',
    docs: 'Vec<Text>'
  },```

Comment (kziemianek) [reply packages/types/src/interfaces/contractsAbi/definitions.ts:192]: @jacogr what do you think about my proposition? If it's still not correct, I kindly ask you to explicitly point how to name these types 🙃

**Comment by peetzweg** [comment]
Would it be smart to adapt the pr to include this change from ink 4.2.0 as well? Otherwise we bump the metadata type version yet again in a short amount of time. 🤔 

https://github.com/polkadot-js/api/issues/5637

**Pull request closed without merging**